src/packages/core/components/ufm-render/ufm-render.element.ts

```diff
@@ -0,0 +1,125 @@
+import { css, customElement, nothing, property, unsafeHTML, until } from '@umbraco-cms/backoffice/external/lit';
+import { DOMPurify } from '@umbraco-cms/backoffice/external/dompurify';
+import { Marked } from '@umbraco-cms/backoffice/external/marked';
+import { createExtensionApi } from '@umbraco-cms/backoffice/extension-api';
+import { ufm } from '@umbraco-cms/backoffice/ufm';
+import type { UfmComponentBase } from '../../../ufm/components/ufm-component-base.js';
+import { UmbUfmRenderContext } from './ufm-render.context.js';
+import { UmbLitElement } from '@umbraco-cms/backoffice/lit-element';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import type { ManifestUfmComponent } from '@umbraco-cms/backoffice/extension-registry';
+
+const UmbDomPurify = DOMPurify(window);
+const UmbDomPurifyConfig: DOMPurify.Config = {
+	USE_PROFILES: { html: true },
+	CUSTOM_ELEMENT_HANDLING: {
+		tagNameCheck: /^(?:ufm|umb|uui)-.*$/,
+		attributeNameCheck: /.+/,
+		allowCustomizedBuiltInElements: false,
+	},
+};
+
+UmbDomPurify.addHook('afterSanitizeAttributes', function (node) {
+	// set all elements owning target to target=_blank
+	if ('target' in node) {
+		node.setAttribute('target', '_blank');
+	}
+});
+
+const UmbMarked = new Marked({
+	async: true,
+	gfm: true,
+	breaks: true,
+	hooks: {
+		postprocess: (markup) => {
+			return UmbDomPurify.sanitize(markup, UmbDomPurifyConfig) as string;
+		},
+	},
+});
+
+const elementName = 'umb-ufm-render';
+
+@customElement(elementName)
+export class UmbUfmRenderElement extends UmbLitElement {
+	#context = new UmbUfmRenderContext(this);
+
+	@property({ type: Boolean })
+	inline = false;
+
+	@property()
+	public set markdown(value: string | undefined) {
+		this.#markdown = value ?? '';
+	}
+	public get markdown(): string | undefined {
+		return this.#markdown;
+	}
+	#markdown: string = '';
+
+	@property({ attribute: false })
+	public set value(value: string | unknown | undefined) {
+		this.#context.setValue(value);
+	}
+	public get value(): string | unknown | undefined {
+		return this.#context.getValue();
+	}
+
+	constructor() {
+		super();
+		this.#observe();
+	}
+
+	async #observe() {
+		this.observe(umbExtensionsRegistry.byType('ufmComponent'), async (manifests: Array<ManifestUfmComponent>) => {
```

Comment:
Here we should use some of the Extension Initializers, which would be more performant as they would only spin up new APIs. :-)

Reply:
Thanks! Implemented in https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/2045/files#diff-204dc0a9aa5ce0ecdf69b1551fc0db0ae55c02e7a1d40f7052d2d0fc76030927R71